AK/SinglyLinkedList.h

```diff
@@ -81,9 +81,15 @@ class SinglyLinkedList {
 public:
     SinglyLinkedList() = default;
     SinglyLinkedList(const SinglyLinkedList& other) = delete;
-    SinglyLinkedList(SinglyLinkedList&&) = default;
+    SinglyLinkedList(SinglyLinkedList&& other)
+        : m_head { move(other.m_head) }
+        , m_tail { move(other.m_tail) }
+    {
+        m_head = nullptr;
+        m_tail = nullptr;
```

Comment:
This leaks the list. I think you meant to set other.m_head and other.m_tail to nullptr?